js/script.js

```diff
@@ -155,12 +157,15 @@ const editTask = (taskId) => {
 };
 
 const renderTasks = () => {
+  const searchText = $searchInput.value.toLowerCase();
   $taskList.innerHTML = "";
 
-  tasks.forEach((task) => {
-    const taskCard = createTaskCard(task);
-    $taskList.appendChild(taskCard);
-  });
+  tasks
+    .filter((task) => task.title.toLowerCase().includes(searchText))
+    .forEach((task) => {
+      const taskCard = createTaskCard(task);
+      $taskList.appendChild(taskCard);
+    });
```

Comment:
```suggestion
  const searchedTasks =  tasks.filter((task) => task.title.toLowerCase().includes(searchText));
  
   searchedTasks.forEach((task) => {
      const taskCard = createTaskCard(task);
      $taskList.appendChild(taskCard);
    });
```

Instead of method chaining, use an extra constant.